### PR TITLE
feat(modal): add modal and overlay components

### PR DESCRIPTION
**Pull request by GuillaumeBAECHLER**
Add Overlay and Modal components.
Rewrite App component to a non-functional component (so we can access context).

### CODE REVIEW

**Review by AlexandreBarbier — APPROVED** [review]

File: packages/mikado_reborn/src/components/Overlay/Overlay.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  @Prop({ type: Boolean, default: false })
+  readonly keepOnCick!: boolean;
+
+  mounted(): void {
+    const app = this.$app;
+    app.$el.insertBefore(this.$el, app.$el.children[0]);
+  }
+
+  click(): void {
+    if (!this.keepOnCick) {
```
Comment (hhochart29) [inline packages/mikado_reborn/src/components/Overlay/Overlay.vue:49]:
```suggestion
    if (!this.keepOnClick) {
```

File: packages/mikado_reborn/src/components/Overlay/Overlay.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  @Prop({
+    type: String,
+    validator: (value: string): boolean => Object.values(colors).includes(value),
+    default: 'dark',
+  })
+  readonly color!: (keyof typeof colors);
+
+  @Prop({ type: Boolean, default: false })
+  readonly keepOnCick!: boolean;
```
Comment (hhochart29) [inline packages/mikado_reborn/src/components/Overlay/Overlay.vue:41]:
```suggestion
  readonly keepOnClick!: boolean;
```

File: packages/mikado_reborn/src/components/Modal/Modal.vue [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  readonly size!: (keyof typeof sizes);
+
+  @Prop({ type: Boolean, default: false })
+  readonly slim!: boolean;
+
+  mounted(): void {
+    const app = this.$app;
+    app.$el.insertBefore(this.$el, app.$el.children[0]);
+    document.addEventListener('mousedown', this.onClickOutside);
+    document.addEventListener('keydown', this.keydownHandler);
```
Comment (hhochart29) [inline packages/mikado_reborn/src/components/Modal/Modal.vue:67]: clean 👌

File: packages/mikado_reborn/src/components/Modal/Modal.vue [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    document.removeEventListener('mousedown', this.onClickOutside);
+    document.removeEventListener('keydown', this.keydownHandler);
+  }
+
+  onClickOutside(event: MouseEvent): void {
+    const target = (event.target as Node | null);
+    if (!target) {
+      return;
+    }
+    const isClickInModal = this.$el.contains(event.target as Node);
```
Comment (hhochart29) [inline packages/mikado_reborn/src/components/Modal/Modal.vue:80]: 💯

**Review by hhochart29 — APPROVED** [review]
Top ! bien joué pour la gestion du click-outside :)

File: packages/mikado_reborn/src/components/Modal/Modal.vue (hunk 3 above)
Comment (GuillaumeBAECHLER) [reply packages/mikado_reborn/src/components/Modal/Modal.vue:67]: thx 😄

**Pull request merged**